### PR TITLE
Added support for filtering topics via regular expressions

### PR DESCRIPTION
**Pull request by esteve**
This PR adds a new option `--topics-regex` that allows filtering topics via regular expressions.

### CODE REVIEW

**Comment by esteve** [comment]
I'm debugging an issue with the player and regular expressions. The following causes `ros2 bag play` to throw an exception:

```shell

$ ros2 bag play . --topics-regex /lidar_front/points_raw
[INFO] [1656517413.919186963] [rosbag2_storage]: Opened database './rosbag2_2020_09_23-15_58_07.db3' for READ_ONLY.
[INFO] [1656517413.919316194] [rosbag2_player]: Set rate to 1
[INFO] [1656517413.925218047] [rosbag2_player]: Adding keyboard callbacks.
[INFO] [1656517413.925259647] [rosbag2_player]: Press SPACE for Pause/Resume
[INFO] [1656517413.925286814] [rosbag2_player]: Press CURSOR_RIGHT for Play Next Message
[INFO] [1656517413.925312879] [rosbag2_player]: Press CURSOR_UP for Increase Rate 10%
[INFO] [1656517413.925341094] [rosbag2_player]: Press CURSOR_DOWN for Decrease Rate 10%
[INFO] [1656517413.925446615] [rosbag2_player]: Playback until timestamp: -1
[INFO] [1656517413.925790959] [rosbag2_storage]: Opened database './rosbag2_2020_09_23-15_58_07.db3' for READ_ONLY.
[ERROR] [1656517414.026479569] [rosbag2_player]: Failed to play: Error when preparing SQL statement 'SELECT data, timestamp, topics.name, messages.id FROM messages JOIN topics ON messages.topic_id = topics.id WHERE ((topics.name REGEXP '/lidar_front/points_raw')) AND (((timestamp = 1600901887524769241) AND (messages.id >= 0)) OR (timestamp > 1600901887524769241)) ORDER BY messages.timestamp, messages.id;'. SQLite error: (1): SQL logic error
```

however, when issuing that same query directly via the sqlite3 command line interface, I get the correct results.

**Comment by esteve** [comment]
This now works, you can pass multiple arguments to `--topics-regex` and `ros2 bag play` will only play the topics that match the regular expressions (e.g. `--topics-regex /lidar_.*/points_raw`)

**Review by jhdcs — APPROVED** [review]
This looks okay to me, though I wouldn't mind a second opinion from another reviewer.

**Comment by MichaelOrlov** [comment]
> This looks okay to me, though I wouldn't mind a second opinion from another reviewer.

I will review it in a couple days.

**Review by MichaelOrlov — CHANGES_REQUESTED** [review]
@esteve Overall looks good to me.  
Among one nitpick with pointers initialization.
Also could you please write a few unit tests to cover new `topics_regex` API ?

File: rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_wrapper.cpp [hunk 1]
```diff
@@ -138,4 +164,11 @@ sqlite3 * SqliteWrapper::get_database()
   return db_ptr;
 }
 
+void SqliteWrapper::initialize_application_functions()
+{
+  sqlite3_create_function(
+    db_ptr, "regexp", 2, SQLITE_ANY, 0,
+    &sqlite3_application_functions::sqlite_regexp, 0, 0);
```
[the remark this answers is not in the document]
Comment (esteve) [reply rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_wrapper.cpp]: Done in b09be60

**Comment by MichaelOrlov** [comment]
Gist: https://gist.githubusercontent.com/MichaelOrlov/4902ef223bb586d0f7311fd7508bc496/raw/28022ca5ed769da33c9731136f0d03173f8dec13/ros2.repos
BUILD args:  --packages-above-and-dependencies ros2bag rosbag2_py rosbag2_storage rosbag2_storage_default_plugins rosbag2_transport rosbag2_tests
TEST args:  --packages-above ros2bag rosbag2_py rosbag2_storage rosbag2_storage_default_plugins rosbag2_transport rosbag2_tests
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/10473
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16917/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11483/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/17316/)

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/ros-2-tsc-meeting-minutes-8-18-2022/27050/1

**Pull request merged**